WordPress/Classes/Services/MediaLibrary.swift

```diff
@@ -1,7 +1,192 @@
 import Foundation
 
-/// Encapsulates interfacing with Media objects and their assets, whether locally on disk or remotely available.
+/// Encapsulates interfacing with Media objects and their assets, whether locally on disk or remotely.
+///
+/// - Note: Methods with escaping closures will call back via the configured managedObjectContex.performBlock
+///   method and it's corresponding thread.
 ///
 open class MediaLibrary: LocalCoreDataService {
-    // Implementation pending for partial PR reviews.
+
+    // MARK: - Instance methods
+
+    /// Creates a Media object with an absoluteLocalURL for a PHAsset's data, asynchronously.
+    ///
+    /// - parameter onMedia: Called if the Media was successfully created and the asset's data exported to an absoluteLocalURL.
+    /// - parameter onError: Called if an error was encountered during creation, error convertible to NSError with a localized description.
+    ///
+    public func makeMediaWith(blog: Blog, asset: PHAsset, onMedia: @escaping (Media) -> (), onError: ((Error) -> ())?) {
+        DispatchQueue.global(qos: .default).async {
```

Comment:
Yeah I think so, let's go ahead and wait on this though as I'd like to test that change along with the integration, if I can.